scripts/llama_summary.py

```diff
@@ -0,0 +1,181 @@
+    Generate a summary for a meeting using GitHub's AI model.
+
+    Args:
+        summary_data: Summary data dictionary
+
+    Returns:
+        Generated summary text
+    """
+    if not GITHUB_TOKEN:
+        print(
+            "Warning: GITHUB_TOKEN environment variable not set. Using placeholder summary."
+        )
+        return "[PLACEHOLDER] Unable to generate AI summary - no GitHub token provided."
+
+    try:
+        # Format the meeting information for the prompt
+        meeting_info = await get_summary_text(summary)
+
+        # Prepare the request payload
+        payload = {
+            "messages": [
+                {
+                    "role": "system",
+                    "content": "You are a government meeting summarizer. Your task is to create concise, informative summaries of government meetings.",
+                },
+                {
+                    "role": "user",
+                    "content": f"Please generate a summary for this government meeting. Focus on the key topics discussed, decisions made, and any notable public comments. If exact details aren't available, provide a general summary of what typically happens in this type of meeting.\n\n{meeting_info}",
+                },
+            ],
+            "temperature": 0.7,
+            "max_tokens": 500,
+            "model": MODEL_NAME,
+        }
+
+        # Make the API request
+        headers = {"Content-Type": "application/json", "api-key": GITHUB_TOKEN}
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(
+                GITHUB_API_ENDPOINT, json=payload, headers=headers
+            ) as response:
+                if response.status == 200:
+                    result = await response.json()
+                    if "choices" in result and len(result["choices"]) > 0:
+                        return result["choices"][0]["message"]["content"]
+                    else:
+                        print(f"Error: Unexpected response format: {result}")
+                else:
+                    error_text = await response.text()
+                    print(
+                        f"Error: API request failed with status {response.status}: {error_text}"
+                    )
+
+        # Fallback to placeholder if API call fails
+        return f"[PLACEHOLDER] This meeting ({summary.get('meeting_name')}) summary will be generated automatically."
+
+    except Exception as e:
+        print(f"Error generating summary: {e}")
+        return f"[PLACEHOLDER] Error generating summary: {str(e)}"
+
+
+async def update_summaries(summaries: List[MeetingSummary]) -> List[MeetingSummary]:
+    new_summaries = []
+
+    # Process each summary that needs updating
+    for summary in summaries:
+        print(
+            f"Generating summary for meeting: {summary.meeting.meeting} ({summary.meeting.date})"
+        )
+
+        # Generate the AI summary
+        ai_summary = await summarize_with_llama(summary)
```

Comment:
suggestion (non-blocking): If we're using `aiohttp` to do these concurrently, shouldn't we remove the `await` from this call? So the process can do all the `summaries` concurrently? Or am I forgetting how async http and concurrency works?

Reply:
Yes, a gather() does make sense here.